pkg/sql/crdb_internal.go

```diff
@@ -4895,51 +4900,154 @@ CREATE TABLE crdb_internal.kv_store_status (
 	},
 }
 
-// crdbInternalPredefinedComments exposes the predefined
-// comments for virtual tables. This is used by SHOW TABLES WITH COMMENT
-// as fall-back when system.comments is silent.
-// TODO(knz): extend this with vtable column comments.
-//
-// TODO(tbg): prefix with node_.
-var crdbInternalPredefinedCommentsTable = virtualSchemaTable{
-	comment: `comments for predefined virtual tables (RAM/static)`,
+var crdbInternalCatalogDescriptorTable = virtualSchemaTable{
+	comment: `like system.descriptor but overlaid with in-txn in-memory changes and including virtual objects`,
 	schema: `
-CREATE TABLE crdb_internal.predefined_comments (
-	TYPE      INT,
-	OBJECT_ID INT,
-	SUB_ID    INT,
-	COMMENT   STRING
+CREATE TABLE crdb_internal.kv_catalog_descriptor (
+  id            INT NOT NULL,
+  descriptor    JSON NOT NULL
 )`,
 	populate: func(
 		ctx context.Context, p *planner, dbContext catalog.DatabaseDescriptor, addRow func(...tree.Datum) error,
 	) error {
-		// NB if ever anyone were to extend this table to carry column
-		// comments, make sure to update pg_catalog.col_description to
-		// retrieve those comments.
-		tableCommentKey := tree.NewDInt(tree.DInt(catalogkeys.TableCommentType))
-		vt := p.getVirtualTabler()
-		vEntries := vt.getSchemas()
-		vSchemaNames := vt.getSchemaNames()
+		all, err := p.Descriptors().GetAll(ctx, p.Txn())
+		if err != nil {
+			return err
+		}
+		// Delegate privilege check to system table.
+		{
+			sysTable := all.LookupDescriptor(systemschema.DescriptorTable.GetID())
+			if p.CheckAnyPrivilege(ctx, sysTable) != nil {
```

Comment:
Should this be CheckAnyPrivilege or should it be something else? Namely select

Reply:
Oh, good point. I shouldn't have blindly copied that.